src/sentry/utils/performance_issues/detectors/sql_injection_detector.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+import hashlib
+from typing import Any
+
+from sentry.issues.grouptype import DBInjectionVulnerabilityGroupType
+from sentry.issues.issue_occurrence import IssueEvidence
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.utils.performance_issues.base import (
+    DetectorType,
+    PerformanceDetector,
+    fingerprint_span,
+    get_notification_attachment_body,
+)
+from sentry.utils.performance_issues.performance_problem import PerformanceProblem
+from sentry.utils.performance_issues.types import Span
+
+MAX_EVIDENCE_VALUE_LENGTH = 10_000
+
+SQL_KEYWORDS = [
+    "SELECT",
+    "WHERE",
+    "AND",
+    "OR",
+    "NOT",
+    "IN",
+    "LIKE",
+    "LIMIT",
+    "ORDER BY",
+    "GROUP BY",
+    "HAVING",
+    "DISTINCT",
+    "JOIN",
+    "ON",
+    "AS",
+    "CASE",
+    "WHEN",
+    "THEN",
+    "ELSE",
+    "END",
+    "UNION",
+    "ALL",
+    "ANY",
+    "SOME",
+    "EXISTS",
+]
+
+
+class SQLInjectionDetector(PerformanceDetector):
+    __slots__ = "stored_problems"
+
+    type = DetectorType.SQL_INJECTION
+    settings_key = DetectorType.SQL_INJECTION
+
+    def __init__(self, settings: dict[DetectorType, Any], event: dict[str, Any]) -> None:
+        super().__init__(settings, event)
+
+        self.stored_problems = {}
+
+        query_string = event.get("request", {}).get("query_string", None)
+        if not query_string:
+            return
+
+        self.user_inputs = [
+            query_value[1] for query_value in query_string if self.keep_query_value(query_value)
+        ]
+
+    def visit_span(self, span: Span) -> None:
+        if not SQLInjectionDetector.is_span_eligible(span):
+            return
+
+        description = span.get("description")
+        fingerprint = fingerprint_span(span)
+        op = span.get("op", None)
+        spans_involved = [span["span_id"]]
+
+        if any(user_input in description for user_input in self.user_inputs):
+            for user_input in self.user_inputs:
+                fingerprint_description = span.get("description")
```

Comment:
nit: The right side should be `description` so check in the loop and the fingerprinting here never diverge unintentionally  if other people make changes later on.